### ISSUE TITLE
Feature Request: Block a user

### ISSUE THREAD

**Issue by desirottman** [thepracticaldev/dev.to#3145]
**Is your feature request related to a problem? Please describe.**
It seems like if there is a way currently to block another user on the platform, it’s not intuitive. DEV is a happy place (especially when compared to the rest of the internet!) but there are still many risks and reason someone might want to block another user.

**Describe the solution you'd like**
An easy-to-find button on a user’s profile that says “block.” Maybe there’s a form that requests feedback for the admin team about why there’s a block (maybe that person is being abusive outside the platform and the blocker just doesn’t want to see their activity on DEV, or maybe the blocker doesn’t want/need to give feedback)

**Describe alternatives you've considered**
Contacting an admin to request a block could be one alternative, but that might make the blocker feel they’d have to explain or justify their request, making them less likely to request the block. Requiring that much “extra work” might also drive them away from requesting the block.

**Additional context**
tl;dr: DEV.to is a happy place, but Bad Guys on the internet will follow us around anywhere they can find us, let’s help keep users safe and protect them from content or people they don’t want to see!

**Comment by mariocsee** [comment]
Hey @desirottman, thanks for submitting this feature request! Preventing abuse and harassment in dev.to is very important for us, and blocking would definitely put more immediate control into users to protect themselves. Currently, we only have blocking on Connect, which we shipped alongside Open Inbox and prevents opening direct chat channels between you and the blocked user. 

In our meeting earlier, we briefly discussed some details on the implementation of blocking. 

Given above, a user you blocked would be:
- unable to follow you
- unable to react and comment on your published articles
- unable to reply to your comments on other threads/articles
- unable to open a direct chat channel with you (automatically block on Connect as well)

You would also be:
- unable to follow the blocked user
- unable to see their posts on your feed
- capable of un-blocking a user
- unable to interact with them (articles/comments/chat).

Since dev.to is open for those not signed up, they would be able to view articles or comments should the blocked user sign out, but of course, they would not be able to interact unless they signed into an unblocked account.

**Comment by desirottman** [comment]
thanks for the feedback! I know that a true "full" block isn't possible anywhere - Twitter is the same way, if a blocked user is logged out, they can still see your public content - it was just something I'd been thinking about.

thanks so much for taking it seriously and talking about it with the team!

**Comment by rhymes** [comment]
I believe this was addressed with https://github.com/thepracticaldev/dev.to/pull/4411

You can see the changelog post here: https://dev.to/devteam/changelog-block-tools-for-user-safety-4no3

Going to close it, @desirottman, let us know if it's not the case and I will reopen
